### ISSUE TITLE
Great work and a few warnings to address

### ISSUE THREAD

**Issue by Innary** [IDEA-Research/GroundingDINO#6]
I appreciate the fantastic work on this project! The paper and implementation are outstanding. Just a heads-up, there are a few PyTorch-related warnings (e.g., floordiv, meshgrid) in the code.

```
/home/yanai-lab/foo/conda_env/gdino/lib/python3.10/site-packages/torch/functional.py:478: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at  /opt/conda/conda-bld/pytorch_1659484808560/work/aten/src/ATen/native/TensorShape.cpp:2894.)  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]final text_encoder_type: bert-base-uncased_IncompatibleKeys(missing_keys=[], unexpected_keys=['label_enc.weight'])/home/yanai-lab/foo/conda_env/gdino/lib/python3.10/site-packages/torch/utils/checkpoint.py:25: UserWarning: None of the inputs have requires_grad=True. Gradients will be None  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")/host/space0/GroundingDINO/groundingdino/models/GroundingDINO/backbone/position_encoding.py:114: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').  dim_tx = self.temperatureW ** (2 * (dim_tx // 2) / self.num_pos_feats)/host/space0/GroundingDINO/groundingdino/models/GroundingDINO/backbone/position_encoding.py:118: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').  dim_ty = self.temperatureH ** (2 * (dim_ty // 2) / self.num_pos_feats)/host/space0/GroundingDINO/groundingdino/models/GroundingDINO/utils.py:209: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  dim_t = 10000 ** (2 * (dim_t // 2) / 128)
```

Here is my current environment for running your code:
```
Package            Version    Editable project location
------------------ ---------- --------------------------------------------------
addict             2.4.0 
brotlipy           0.7.0 
certifi            2022.12.7
cffi               1.15.1
chardet            5.1.0 
charset-normalizer 3.1.0 
click              8.1.3 
contourpy          1.0.7 
cryptography       39.0.1
cycler             0.11.0
filelock           3.10.0
flit_core          3.8.0  
fonttools          4.39.2
groundingdino      0.1.0      /host/space0/xx/GroundingDINO
huggingface-hub    0.13.3                                                                                                                                            
idna               3.4
joblib             1.2.0
kiwisolver         1.4.4
matplotlib         3.7.1
mkl-fft            1.3.1
mkl-random         1.2.2
mkl-service        2.4.0
numpy              1.24.2
opencv-python      4.7.0.72
packaging          21.3
Pillow             9.4.0
pip                23.0.1
pycocotools        2.0.6
pycparser          2.21
pyOpenSSL          23.0.0
pyparsing          3.0.9
PySocks            1.7.1
python-dateutil    2.8.2
PyYAML             6.0
regex              2022.10.31
requests           2.28.2
sacremoses         0.0.53
setuptools         65.6.3
six                1.16.0
timm               0.6.12
tokenizers         0.10.3
torch              1.12.1
torchaudio         0.12.1
torchvision        0.13.1
tqdm               4.65.0
transformers       4.5.1
typing_extensions  4.4.0
urllib3            1.26.15
wheel              0.38.4
yapf               0.32.0
```

**Comment by austinmw** [comment]
Any idea why `UserWarning: None of the inputs have requires_grad=True. Gradients will be None` is occurring?
